prototypes/technology/trainassembly-technology.lua

```diff
@@ -0,0 +1,151 @@
+require "util"
+
+local trainTech = util.table.deepcopy(data.raw["technology"]["railway"])
+
+  trainTech.name = "trainassembly-trainTechnology"
+  trainTech.effects = {}
+  trainTech.localised_name = {"technology-name.trainTech"}
+  trainTech.localised_description = {"technology-description.trainTech"}
+
+  data:extend({
+    trainTech,
+  })
+
+for _, trainTechn in pairs ({
```

Comment:
I would change the name trainTechn over to trainRecipe so its more obvious what it represent